### PR TITLE
PLAT-138030: Migrate event-logger sample to sandstone

### PR DESCRIPTION
**Pull request by juwonjeong**
Enact-DCO-1.0-Signed-off-by: Juwon Jeong (juwon.jeong@lge.com)
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Add event-logger sample for sandstone 

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Migrate event-logger sample to sandstone

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
PLAT-138030

### CODE REVIEW

File: samples/event-logger/src/views/MainView.module.less [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
+@background-color: #e6e6e6;
+@hover-background-color:#abaeb3;
+@font-color: #4c5059;
 
 .mainView {
 	display: flex;
-	flex-direction: column;
 	width: 100%;
 	height: 100%;
+	flex-direction: column;
```
Comment (jeonghee27) [inline samples/event-logger/src/views/MainView.module.less]: Why did you change order?

File: samples/event-logger/src/components/InputBoard/Filter/Filter.module.less [hunk 2]
```diff
@@ -1,13 +1,33 @@
...
+		}
+	}
 }
+
+@media (max-width: 1284px) {
+	.eventGroup {
+		flex-direction: row;
+		flex-wrap: wrap;
+	}
+}
```
Comment (jeonghee27) [inline samples/event-logger/src/components/InputBoard/Filter/Filter.module.less:33]: Please add empty line

File: samples/event-logger/src/components/InputBoard/Filter/Filter.module.less [hunk 3]
```diff
@@ -1,13 +1,33 @@
...
 }
 
 .eventGroup {
-	padding: 12px;
+	display: flex;
+	.pickerItem {
+		display: inline-flex;
+		margin-bottom: 30px;
+		align-items: center;
+		line-height: 72px;
```
Comment (jeonghee27) [inline samples/event-logger/src/components/InputBoard/Filter/Filter.module.less:15]: As I checked, we don't need line-height

File: samples/event-logger/src/components/InputBoard/Filter/Filter.module.less [hunk 4]
```diff
@@ -1,13 +1,33 @@
...
-	position: relative;
-	.timerPicker {
-		transform: scale(0.8);
-	}
+.heading {
+	margin-bottom: 0
+}
+
+.timerPicker {
+	transform: scale(0.8);
```
Comment (jeonghee27) [inline samples/event-logger/src/components/InputBoard/Filter/Filter.module.less:6]: let's remove this

File: samples/event-logger/src/components/InputBoard/Filter/Filter.module.less [hunk 5]
```diff
@@ -1,13 +1,33 @@
-.timer {
-	font-size: 24px;
-	line-height: 52px;
-	padding: 0 12px;
-	position: relative;
-	.timerPicker {
-		transform: scale(0.8);
-	}
+.heading {
+	margin-bottom: 0
```
Comment (jeonghee27) [inline samples/event-logger/src/components/InputBoard/Filter/Filter.module.less:2]: Remove it

File: samples/event-logger/src/components/InputBoard/Filter/Filter.module.less [hunk 6]
```diff
@@ -1,13 +1,33 @@
...
+		&.small {
+			width: 480px;
+		}
+		&.large {
+			max-width: 840px;
+		}
+	}
 }
+
+@media (max-width: 1284px) {
```
Comment (jeonghee27) [inline samples/event-logger/src/components/InputBoard/Filter/Filter.module.less:28]: Please add description for it

File: samples/event-logger/src/components/InputBoard/Filter/Filter.module.less (hunk 2 above)
Comment (juwonjeong) [reply samples/event-logger/src/components/InputBoard/Filter/Filter.module.less]: It's fixed.

File: samples/event-logger/src/components/InputBoard/Filter/Filter.module.less (hunk 3 above)
Comment (juwonjeong) [reply samples/event-logger/src/components/InputBoard/Filter/Filter.module.less]: It's removed.

File: samples/event-logger/src/components/InputBoard/Filter/Filter.module.less (hunk 4 above)
Comment (juwonjeong) [reply samples/event-logger/src/components/InputBoard/Filter/Filter.module.less]: It's removed.

File: samples/event-logger/src/components/InputBoard/Filter/Filter.module.less (hunk 5 above)
Comment (juwonjeong) [reply samples/event-logger/src/components/InputBoard/Filter/Filter.module.less]: It's removed.

File: samples/event-logger/src/components/InputBoard/Filter/Filter.module.less (hunk 6 above)
Comment (juwonjeong) [reply samples/event-logger/src/components/InputBoard/Filter/Filter.module.less:20]: It's fixed.

**Review by jeonghee27 — APPROVED** [review]
LGTM

File: samples/event-logger/src/components/InputBoard/Filter/Filter.js [hunk 7]
```diff
@@ -0,0 +1,137 @@
+import {Component} from 'react';
+import {connect} from 'react-redux';
+import PropTypes from 'prop-types';
+import Heading from '@enact/sandstone/Heading';
+import Picker from '@enact/sandstone/Picker';
+import SwitchItem from '@enact/sandstone/SwitchItem';
```
Comment (MikyungKim) [inline samples/event-logger/src/components/InputBoard/Filter/Filter.js:6]: Please re-order these based on our coding rule.

File: samples/event-logger/src/components/InputBoard/Filter/Filter.js [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+	render () {
+		const
+			timergroup = ['3 sec', '5 sec', '10 sec'],
+			eventItems = eventCategory.map((e, i) => {
+				const handler = this.handleEventCategory(i);
+				return (
+					<SwitchItem
+						className={[css.switchItem, css.small]}
+						key={i}
+						inline
```
Comment (MikyungKim) [inline samples/event-logger/src/components/InputBoard/Filter/Filter.js:55]:
```suggestion
						inline
						key={i}
```

File: samples/event-logger/src/components/InputBoard/InputBoard.js [hunk 9]
```diff
@@ -0,0 +1,242 @@
+import {connect} from 'react-redux';
+import {createRef, Component} from 'react';
+import PropTypes from 'prop-types';
```
Comment (MikyungKim) [inline samples/event-logger/src/components/InputBoard/InputBoard.js:3]:
```suggestion
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {createRef, Component} from 'react';
```

File: samples/event-logger/src/components/Log/Log.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
+import {Component} from 'react';
+import PropTypes from 'prop-types';
+
+import css from './Log.module.less';
+
+class Log extends Component {
```
Comment (MikyungKim) [inline samples/event-logger/src/components/Log/Log.js:6]: Does this component need to be a `class`? Why don't we use `kind`?

File: samples/event-logger/src/components/Log/Logs.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
+import {connect} from 'react-redux';
+import kind from '@enact/core/kind';
+import PropTypes from 'prop-types';
+import Scroller from '@enact/ui/Scroller';
+
+import Log from './Log';
+import css from './Log.module.less';
```
Comment (MikyungKim) [inline samples/event-logger/src/components/Log/Logs.js:7]:
```suggestion

import css from './Log.module.less';
```

File: samples/event-logger/src/index.js [hunk 12]
```diff
@@ -0,0 +1,19 @@
+import {Provider} from 'react-redux';
+import {render} from 'react-dom';
+
+import App from './App';
+import storeFactory from './store/store';
```
Comment (MikyungKim) [inline samples/event-logger/src/index.js:5]:
```suggestion
import storeFactory from './store';
```

File: samples/event-logger/src/components/InputBoard/Filter/Filter.js (hunk 7 above)
Comment (juwonjeong) [reply samples/event-logger/src/components/InputBoard/Filter/Filter.js]: It's fixed.

File: samples/event-logger/src/components/InputBoard/InputBoard.js (hunk 9 above)
Comment (juwonjeong) [reply samples/event-logger/src/components/InputBoard/InputBoard.js]: It's fixed.

File: samples/event-logger/src/components/InputBoard/Filter/Filter.js [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+					showLine
+					size="tiny"
+					spacing="small"
+				>
+					Events
+				</Heading>
+				<div className={css.eventGroup}>
+					{eventItems}
+				</div>
+
```
Comment (MikyungKim) [inline samples/event-logger/src/components/InputBoard/Filter/Filter.js:75]: an empty line?

File: samples/event-logger/src/components/InputBoard/InputBoard.js [hunk 14]
```diff
@@ -0,0 +1,242 @@
...
+import {createRef, Component} from 'react';
+import {connect} from 'react-redux';
+
+import {addEventLog, removeEventLog, updateEventLog} from '../../actions/actions';
```
Comment (MikyungKim) [inline samples/event-logger/src/components/InputBoard/InputBoard.js:5]:
```suggestion
import {addEventLog, removeEventLog, updateEventLog} from '../../actions';
```

File: samples/event-logger/src/components/InputBoard/InputBoard.js [hunk 15]
```diff
@@ -0,0 +1,242 @@
...
+import eventCategory from '../../constants/eventCategory';
+
+import eventRegData from './Event/EventRegistrationData';
+import Filter from './Filter/Filter';
```
Comment (MikyungKim) [inline samples/event-logger/src/components/InputBoard/InputBoard.js:9]:
```suggestion
import Filter from './Filter';
```

File: samples/event-logger/src/views/MainView.js [hunk 16]
```diff
@@ -0,0 +1,24 @@
+import kind from '@enact/core/kind';
+
+import InputBoard from '../components/InputBoard/InputBoard';
```
Comment (MikyungKim) [inline samples/event-logger/src/views/MainView.js:3]:
```suggestion
import InputBoard from '../components/InputBoard';
```

File: samples/event-logger/src/views/MainView.module.less [hunk 17]
```diff
@@ -0,0 +1,24 @@
+@background-color: #e6e6e6;
+@hover-background-color:#abaeb3;
```
Comment (MikyungKim) [inline samples/event-logger/src/views/MainView.module.less:2]:
```suggestion
@hover-background-color: #abaeb3;
```

File: samples/event-logger/src/components/InputBoard/Filter/Filter.module.less [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+	.pickerItem {
+		display: inline-flex;
+		margin-bottom: 30px;
+		align-items: center;
+	}
+	.switchItem {
+		font-size: 48px;
+		&.small {
+			width: 480px;
+		}
```
Comment (MikyungKim) [inline samples/event-logger/src/components/InputBoard/Filter/Filter.module.less:12]: I think we don't need this.

File: samples/event-logger/src/components/InputBoard/Filter/Filter.js [hunk 19]
```diff
@@ -0,0 +1,137 @@
...
+		this.props.onIsSyntheticEventOn(selected);
+	};
+
+	render () {
+		const
+			timergroup = ['3 sec', '5 sec', '10 sec'],
+			eventItems = eventCategory.map((e, i) => {
+				const handler = this.handleEventCategory(i);
+				return (
+					<SwitchItem
```
Comment (MikyungKim) [inline samples/event-logger/src/components/InputBoard/Filter/Filter.js:52]: Let's make all `SwitchItem`s to have `size` prop "small".

**Review by MikyungKim — COMMENTED** [review]
One more thing I'd like to suggest you consider is, making a button to show and hide filters. On mobile, the area of logging seems too small so I think it'd be nide if we have such a thing.
image

File: samples/event-logger/src/components/InputBoard/Filter/Filter.js (hunk 13 above)
Comment (juwonjeong) [reply samples/event-logger/src/components/InputBoard/Filter/Filter.js]: It's removed.

File: samples/event-logger/src/components/InputBoard/InputBoard.js (hunk 14 above)
Comment (juwonjeong) [reply samples/event-logger/src/components/InputBoard/InputBoard.js]: It's fixed.

File: samples/event-logger/src/components/InputBoard/InputBoard.js (hunk 15 above)
Comment (juwonjeong) [reply samples/event-logger/src/components/InputBoard/InputBoard.js]: It's fixed.

File: samples/event-logger/src/views/MainView.js (hunk 16 above)
Comment (juwonjeong) [reply samples/event-logger/src/views/MainView.js]: It's fixed.

File: samples/event-logger/src/views/MainView.module.less (hunk 17 above)
Comment (juwonjeong) [reply samples/event-logger/src/views/MainView.module.less]: It's fixed.

File: samples/event-logger/src/components/InputBoard/Filter/Filter.js (hunk 19 above)
Comment (juwonjeong) [reply samples/event-logger/src/components/InputBoard/Filter/Filter.js:55]: It's added.

File: samples/event-logger/src/components/InputBoard/Filter/Filter.module.less (hunk 18 above)
Comment (juwonjeong) [reply samples/event-logger/src/components/InputBoard/Filter/Filter.module.less:12]: I added such style to keep text aligned even if it becomes 2 lines, not like the picture below.
eventFilter
Is it okay to remain `&.small`? If not I will fix it to remove this style.

**Comment by juwonjeong** [comment]
> One more thing I'd like to suggest you consider is, making a button to show and hide filters. On mobile, the area of logging seems too small so I think it'd be nide if we have such a thing.
> image

As you advised, I added a switch to hide and show the filters as shown below.
buttonFilter

File: samples/event-logger/src/components/InputBoard/Filter/Filter.module.less (hunk 18 above)
Comment (MikyungKim) [reply samples/event-logger/src/components/InputBoard/Filter/Filter.module.less:12]: Well, I prefer not to set the width explicitly. Because when it's focused it looks like below.
image
Let's just remove it. :)

File: samples/event-logger/src/components/InputBoard/InputBoard.js [hunk 20]
```diff
@@ -196,11 +200,22 @@ class InputBoardBase extends Component {
...
+	};
 
 	render () {
 		return (
 			<div>
-				<Filter />
+				<SwitchItem
+					selected={this.state.showFilter}
+					size="small"
+					style={{padding: '0px'}}
```
Comment (MikyungKim) [inline samples/event-logger/src/components/InputBoard/InputBoard.js:213]: Let's remove this :)

File: samples/event-logger/src/components/InputBoard/Filter/Filter.module.less (hunk 18 above)
Comment (juwonjeong) [reply samples/event-logger/src/components/InputBoard/Filter/Filter.module.less]: It's removed

File: samples/event-logger/src/components/InputBoard/InputBoard.js (hunk 20 above)
Comment (juwonjeong) [reply samples/event-logger/src/components/InputBoard/InputBoard.js]: It's removed.

**Review by MikyungKim — APPROVED** [review]
LGTM
